packages/aws-cdk-lib/aws-codepipeline/lib/trigger.ts

```diff
@@ -111,16 +204,45 @@ export class Trigger {
+      }
+
       if (pushFilter !== undefined && pushFilter.length > 3) {
         throw new Error(`length of pushFilter for sourceAction with name '${sourceAction.actionProperties.actionName}' must be less than or equal to 3, got ${pushFilter.length}`);
       }
-
       pushFilter?.forEach(filter => {
         if (filter.tagsExcludes && filter.tagsExcludes.length > 8) {
-          throw new Error(`maximum length of tagsExcludes for sourceAction with name '${sourceAction.actionProperties.actionName}' is 8, got ${filter.tagsExcludes.length}`);
+          throw new Error(`maximum length of tagsExcludes in pushFilter for sourceAction with name '${sourceAction.actionProperties.actionName}' is 8, got ${filter.tagsExcludes.length}`);
         }
         if (filter.tagsIncludes && filter.tagsIncludes.length > 8) {
-          throw new Error(`maximum length of tagsIncludes for sourceAction with name '${sourceAction.actionProperties.actionName}' is 8, got ${filter.tagsIncludes.length}`);
+          throw new Error(`maximum length of tagsIncludes in pushFilter for sourceAction with name '${sourceAction.actionProperties.actionName}' is 8, got ${filter.tagsIncludes.length}`);
+        }
+      });
+
+      if (pullRequestFilter !== undefined && pullRequestFilter.length > 3) {
+        throw new Error(`length of pullRequestFilter for sourceAction with name '${sourceAction.actionProperties.actionName}' must be less than or equal to 3, got ${pullRequestFilter.length}`);
+      }
+      pullRequestFilter?.forEach(filter => {
+        if (!filter.branchesExcludes && !filter.branchesIncludes) {
+          throw new Error(`must specify branches in pullRequestFilter for sourceAction with name '${sourceAction.actionProperties.actionName}'`);
+        }
+        if (filter.branchesExcludes && filter.branchesExcludes.length > 8) {
+          throw new Error(`maximum length of branchesExcludes in pullRequestFilter for sourceAction with name '${sourceAction.actionProperties.actionName}' is 8, got ${filter.branchesExcludes.length}`);
+        }
+        if (filter.branchesIncludes && filter.branchesIncludes.length > 8) {
+          throw new Error(`maximum length of branchesIncludes in pullRequestFilter for sourceAction with name '${sourceAction.actionProperties.actionName}' is 8, got ${filter.branchesIncludes.length}`);
+        }
+        if (filter.filePathsExcludes && filter.filePathsExcludes.length > 8) {
+          throw new Error(`maximum length of filePathsExcludes in pullRequestFilter for sourceAction with name '${sourceAction.actionProperties.actionName}' is 8, got ${filter.filePathsExcludes.length}`);
+        }
+        if (filter.filePathsIncludes && filter.filePathsIncludes.length > 8) {
+          throw new Error(`maximum length of filePathsIncludes in pullRequestFilter for sourceAction with name '${sourceAction.actionProperties.actionName}' is 8, got ${filter.filePathsIncludes.length}`);
```

Comment:
We're missing validation for `filter.events` to contain 1 to 3 items if specified.

Reply:
If none are included, all events are automatically set. Also, any duplicates will be eliminated.
So I don't think validation is necessary. What do you think?

https://github.com/go-to-k/aws-cdk/blob/7891faa1ae26b91e01f14b87ab4b8a783f4b0ee5/packages/aws-cdk-lib/aws-codepipeline/lib/trigger.ts#L284